src/Temporalio/Client/WorkflowUpdateOptions.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using Temporalio.Api.Enums.V1;
+
+namespace Temporalio.Client
+{
+    /// <summary>
+    /// Options for starting an update on a <see cref="WorkflowHandle" />.
+    /// </summary>
+    /// <remarks>WARNING: Workflow update is experimental and APIs may change.</remarks>
+    public class WorkflowUpdateOptions : ICloneable
+    {
+        /// <summary>
+        /// Gets or sets the unique identifier for the update. This is optional and is defaulted to
+        /// a GUID if not set. This must be unique within the scope of a workflow execution (i.e.
+        /// namespace + workflow ID + run ID).
+        /// </summary>
+        public string? UpdateID { get; set; }
+
+        /// <summary>
+        /// Gets or sets the stage to wait for before returning from starting an update.
+        /// </summary>
+        public UpdateWorkflowExecutionLifecycleStage WaitForStage { get; set; } = UpdateWorkflowExecutionLifecycleStage.Completed;
```

Comment:
After discussion with team, we may change this to default to unspecified or accepted

Reply:
Right, this is based on the team discussion that happened after I made this comment. Next commit will update this.

Reply:
Updated.